src/Aks/Aks/Models/AdapterHelper.cs

```diff
@@ -0,0 +1,171 @@
+            var targetProps = typeof(Target).GetProperties();
+            foreach (var sourceProp in sourceProps)
+            {
+                var sourceType = sourceProp.GetType();
+                foreach (var targetProp in targetProps)
+                {
+
+                    if (sourceProp.Name == targetProp.Name)
+                    {
+                        object result;
+                        result = sourceProp.GetValue(source);
+                        if (targetProp.PropertyType != sourceProp.PropertyType && sourceProp.GetValue(source) != null)
+                        {
+                            Type type = typeof(AdapterHelper<,>).MakeGenericType(sourceProp.PropertyType, targetProp.PropertyType);
+                            var adapter = Activator.CreateInstance(type, null);
+
+                            MethodInfo method;
+
+                            if (sourceProp.GetValue(source) is IList)
+                                method = type.GetMethod("AdaptList");
+                            else if (sourceProp.GetValue(source) is IDictionary)
+                                method = type.GetMethod("AdaptDict");
+                            else if (sourceProp.GetValue(source).GetType().IsEnum)
+                                method = type.GetMethod("AdaptEnum");
+                            else
+                                method = type.GetMethod("Adapt");
+
+                            result = method.Invoke(adapter,
+                                BindingFlags.Static | BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public,
+                                new PSBinder(),
+                                new object[] { sourceProp.GetValue(source) },
+                                CultureInfo.CurrentCulture);
+                        }
+                        if (targetProp.CanWrite)
+                        {
+                            targetProp.SetValue(targetItem, result);
+                        }
+                        else
+                        {
+                            int layer = 5;
+                            bool flag = true;
+                            Type TargetType = targetItem.GetType();
+                            while (layer-- >= 0 && flag)
+                            {
+                                try
+                                {
+                                    flag = false;
+                                    TargetType.InvokeMember(targetProp.Name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.SetProperty | BindingFlags.Instance, null, targetItem, new object[] { result });
+                                }
+                                catch (Exception)
+                                {
+                                    TargetType = TargetType.BaseType;
+                                    flag = true;
+                                }
+                            }
```

Comment:
I think you can use `GetProperty` first instead of `try catch`. And no need to set a layer.

Reply:
Updated.